src/figuresClasses.ts

```diff
@@ -1,11 +1,74 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error(`These sides can not be a less than zero`);
+    }
+
+    if (this.a + this.b <= this.c) {
+      throw new Error(
+        `The sides ${this.a} and ${this.b} can not be less ${this.c}`,
+      );
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const result = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return +result.toFixed(2);
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  shape: 'circle' = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error(`The radius can't be a less zero`);
+    }
+  }
 
-export class Rectangle implements Figure {}
+  getArea(): number {
+    return Math.floor(Math.PI * Math.pow(this.radius, 2) * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: 'rectangle' = 'rectangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public width: number,
+    public height: number,
+  ) {
+    if (this.width <= 0 || this.height <= 0) {
+      throw new Error(`Error`);
```

Comment:
The error message `Error` is too generic. It should specify the issue, such as `Width and height must be greater than zero.`